exporter/otlpexporter/otlp_test.go

```diff
@@ -677,3 +692,202 @@ func TestSendLogData(t *testing.T) {
 	require.Equal(t, len(md.Get("User-Agent")), 1)
 	require.Contains(t, md.Get("User-Agent")[0], "Collector/1.2.3test")
 }
+
+// TestSendArrowTracesNotSupported tests a successful OTLP export w/
+// and without Arrow, w/ WaitForReady and without.
+func TestSendArrowTracesNotSupported(t *testing.T) {
+	// With arrow unsupported
+	t.Run("Service Unavailable", func(t *testing.T) {
+		t.Run("WaitForReady true", func(t *testing.T) {
+			testSendArrowTraces(t, true, false)
+		})
+		t.Run("WaitForReady false", func(t *testing.T) {
+			testSendArrowTraces(t, false, false)
+		})
+	})
+	// With arrow supported
+	t.Run("Service Available", func(t *testing.T) {
+		t.Run("WaitForReady true", func(t *testing.T) {
+			testSendArrowTraces(t, true, true)
+		})
+		t.Run("WaitForReady false", func(t *testing.T) {
+			testSendArrowTraces(t, false, true)
+		})
+	})
+}
+
+func testSendArrowTraces(t *testing.T, clientWaitForReady, streamServiceAvailable bool) {
+	// Start an OTLP-compatible receiver.
+	ln, err := net.Listen("tcp", "127.0.0.1:")
+	require.NoError(t, err, "Failed to find an available address to run the gRPC server: %v", err)
```

Comment:
These net tests can sometimes run in to issues with CI/CD, have you noticed any potential issues with finding the address:port?

Reply:
I haven't, but I know the mainline collector repo is struggling with this sort of issue in general.